url/src/lib.rs

```diff
@@ -139,6 +139,7 @@ url = { version = "2", features = ["debugger_visualizer"] }
     feature = "debugger_visualizer",
     debugger_visualizer(natvis_file = "../../debug_metadata/url.natvis")
 )]
+#![cfg_attr(all(target_os = "wasi", target_env = "p2"), feature(wasip2))]
```

Comment:
Could you add a comment to this saying specifically "We use the `wasi` OsStrExt, and that is conditionally feature gated to be unstable on wasip2" so folks know why this is there.

Potentially link to https://github.com/rust-lang/rust/issues/130323

Reply:
Done, let me know if you want me to edit the wording / feel free to edit directly. FWIW, I agree with you that this feels unnecessary